azure-iot-device/azure/iot/device/iothub/edge_hsm.py

```diff
@@ -1,80 +1,28 @@
 # Copyright (c) Microsoft Corporation. All rights reserved.
 # Licensed under the MIT License. See License.txt in the project root for
 # license information.
 # --------------------------------------------------------------------------
 
-import os
-import base64
+import logging
 import json
-import six.moves.urllib as urllib
+import base64
 import requests
 import requests_unixsocket
-import logging
-from .base_renewable_token_authentication_provider import BaseRenewableTokenAuthenticationProvider
+from six.moves import urllib, http_client
 from azure.iot.device.common.chainable_exception import ChainableException
-from azure.iot.device.product_info import ProductInfo
+from azure.iot.device.common.auth.signing_mechanism import SigningMechanism
+from azure.iot.device import product_info
 
 requests_unixsocket.monkeypatch()
-
 logger = logging.getLogger(__name__)
 
 
 class IoTEdgeError(ChainableException):
     pass
 
 
-class IoTEdgeAuthenticationProvider(BaseRenewableTokenAuthenticationProvider):
-    """An Azure IoT Edge Authentication Provider.
-
-    This provider creates the Shared Access Signature that would be needed to connenct to the IoT Edge runtime
-    """
-
-    def __init__(
-        self,
-        hostname,
-        device_id,
-        module_id,
-        gateway_hostname,
-        module_generation_id,
-        workload_uri,
-        api_version,
-    ):
-        """
-        Constructor for IoT Edge Authentication Provider
-        """
-
-        logger.info("Using IoTEdge authentication for {%s, %s, %s}", hostname, device_id, module_id)
-
-        super(IoTEdgeAuthenticationProvider, self).__init__(
-            hostname=hostname, device_id=device_id, module_id=module_id
-        )
-
-        self.hsm = IoTEdgeHsm(
-            module_id=module_id,
-            api_version=api_version,
-            module_generation_id=module_generation_id,
-            workload_uri=workload_uri,
-        )
-        self.gateway_hostname = gateway_hostname
-        self.server_verification_cert = self.hsm.get_trust_bundle()
-
-    # TODO: reconsider this design when refactoring the BaseRenewableToken auth parent
-    # TODO: Consider handling the quoting within this function, and renaming quoted_resource_uri to resource_uri
-    def _sign(self, quoted_resource_uri, expiry):
-        """
-        Creates the signature to be inserted in the SAS token
-        :param resource_uri: the resource URI to encode into the token
-        :param expiry: an integer value representing the number of seconds since the epoch 00:00:00 UTC on 1 January 1970 at which the token will expire.
-        :return: The signature portion of the Sas Token.
-
-        :raises: IoTEdgeError if data cannot be signed
-        """
-        string_to_sign = quoted_resource_uri + "\n" + str(expiry)
-        return self.hsm.sign(string_to_sign)
-
-
-class IoTEdgeHsm(object):
+class IoTEdgeHsm(SigningMechanism):
```

Comment:
don't know the solution, but it seems odd that SigningMechanism is in the common.auth sub-package, but this one is not in an auth sub-package

Reply:
Yeah, I'm not thrilled about it, but I didn't really want to make a package just for one module

---
In reply to: [427506098](https://github.com/Azure/azure-iot-sdk-python/pull/555#discussion_r427506098) [](ancestors = 427506098)